File: Capitulo4/secao_4_2/C/sockets/UDP/clienteUDP.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>
#include <arpa/inet.h>

#define SERVER_PORT 9876
#define BUFFER_SIZE 1024

int main() {
    int sockfd;
    struct sockaddr_in server_addr;
    char buffer[BUFFER_SIZE];
    char *mensagem = "sistemas_distribuidos";

    sockfd = socket(AF_INET, SOCK_DGRAM, 0);
    if (sockfd < 0) {
        perror("Erro ao criar socket");
        exit(EXIT_FAILURE);
    }

    memset(&server_addr, 0, sizeof(server_addr));
    server_addr.sin_family = AF_INET;
    server_addr.sin_port = htons(SERVER_PORT);
    server_addr.sin_addr.s_addr = inet_addr("127.0.0.1");

    sendto(sockfd, mensagem, strlen(mensagem), 0, (struct sockaddr *) &server_addr, sizeof(server_addr));
    printf("Enviado: %s\n", mensagem);

    socklen_t len = sizeof(server_addr);
    int n = recvfrom(sockfd, buffer, BUFFER_SIZE, 0, (struct sockaddr *) &server_addr, &len);
    buffer[n] = '\0';
    printf("Recebido: %s\n", buffer);

    close(sockfd);
    return 0;
}

File: Capitulo4/secao_4_2/Python/sockets/UDP/clienteUDP.py
import socket

def main():
    server_address = ('localhost', 9876)
    mensagem = "sistemas_distribuidos"

    sock = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
    try:
        print(f"Enviado: {mensagem}")
        sock.sendto(mensagem.encode(), server_address)
        data, _ = sock.recvfrom(1024)
        print(f"Recebido: {data.decode()}")
    finally:
        sock.close()

if __name__ == "__main__":
    main()
